include/boost/url/impl/segments_view.ipp

```diff
@@ -207,14 +207,12 @@ operator<<(
 {
     auto it = vw.begin();
     auto const end = vw.end();
-    if(! vw.is_absolute())
-        goto skip;
+    if(! vw.is_absolute() && it != end) {
+        os << *it++;
+    }
     while(it != end)
     {
-        os << '/';
-    skip:
-        auto s(*it++);
```

Comment:
This was a segmentation fault when `vw.is_absolute()` was false and the path had 0 segments because we would go to `skip` without checking if `it != end`.

Reply:
it needs a test

Reply:
Yes. I included a test.